ml-agents/mlagents/trainers/ghost/trainer.py

```diff
@@ -38,11 +46,12 @@ def __init__(
         )
 
         self.trainer = trainer
+        self.controller = controller
```

Comment:
There is a cyclic reference between the controller and the ghost trainer. Not sure if we care about those in Python...

Reply:
I know that this isn't ideal but from investigating it doesn't seem to have caused any issues (yet).